x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ConstantNullVector.java

```diff
@@ -54,6 +54,12 @@ public ConstantNullVector filter(int... positions) {
         throw new UnsupportedOperationException("null vector");
     }
 
+    @Override
+    public ConstantNullBlock keepMask(BooleanVector mask) {
+        assert false : "null vector";
+        throw new UnsupportedOperationException("null vector");
```

Comment:
Why do we throw UOE instead of returning a constant null block?

(This seems like something that could blow up in the future, where due to some optimization we end up having a constant null vector in a place where we apply a mask to the vector.)

Reply:
Vectors can never contain `null` anyway. This exists to make the compiler happy. The class level javadoc mentions this - it's mostly so we can return something the implements all of the appropriate interfaces from `ConstantNullBlock`. If there's a way to delete this class I'd be happy to.